source/extensions/filters/listener/tls_inspector/tls_inspector.h

```diff
@@ -20,9 +20,7 @@ namespace TlsInspector {
  * All stats for the TLS inspector. @see stats_macros.h
  */
 #define ALL_TLS_INSPECTOR_STATS(COUNTER)                                                           \
-  COUNTER(connection_closed)                                                                       \
```

Comment:
Are these documented? If so do the docs need to be removed? Also, do we have generic stats for these events on the listener somewhere? I don't think we should lose stats that people may be relying on. If stats move around we will need to have a clear release note about the moves.

Reply:
got it, I checked, there are no generic stats for these events indeed.

`connection_closed` is for remove close.
`read_error` is for `recv` failed.

I can add them to the listener stats, and add doc for them.

Reply:
I added two new stats under listener scope
https://github.com/envoyproxy/envoy/pull/17395/files#diff-f58a327d2818cf58cf9e05a18e8acafee067e616b911a0e805f7b1732a61c7e4R18-R19